#717 Release 25.8.0 (#718)
twisted/towncrier: NEWS.rst

.. towncrier release notes start

towncrier 25.8.0rc2 (2025-08-15)
================================

Features
--------

- Support was added for Mercurial SCM. (`#394 <https://github.com/twisted/towncrier/issues/394>`_)

> - The `towncrier check` command now has a `--staged` flag to inspect the files staged for commit when checking for a news fragment: useful in a pre-commit hook (`#676 <https://github.com/twisted/towncrier/issues/676>`_)

Reviewer: ```suggestion
- The :command:`towncrier check` command now has a ``--staged`` flag to inspect the files staged for commit when checking for a news fragment: useful in a pre-commit hook (`#676 <https://github.com/twisted/towncrier/issues/676>`_)
```

> - When used with an `pyproject.toml` file, when no explicit values are

Reviewer: ```suggestion
- When used with an :file:`pyproject.toml` file, when no explicit values are
```

  defined for [tool.towncrier.name|package] they will now fallback to

>   the value of [project.name]. (`#687 <https://github.com/twisted/towncrier/issues/687>`_)

Reviewer: ```suggestion
  defined for ``[tool.towncrier.name|package]`` they will now fallback to
  the value of ``[project.name]``. (`#687 <https://github.com/twisted/towncrier/issues/687>`_)
```

> - More simple configuration for Keep a Changelog style changelogs (`#691 <https://github.com/twisted/towncrier/issues/691>`_)

Reviewer: This entry isn't very clear from the PoV of an end-user. Reading it in the change log, I don't understand what the effect is for me.

> - More simple configuration for Keep a Changelog style changelogs (`#691 <https://github.com/twisted/towncrier/issues/691>`_)

Author: ```suggestion
- The `directory` configuration option is no longer required.
  It can now be generated based on value from the `name` configuration.
  (`#691 <https://github.com/twisted/towncrier/issues/691>`_)
```

> - Added support for Python 3.13 and removed support for Python 3.8. (`#700 <https://github.com/twisted/towncrier/issues/700>`_)

Reviewer: This should be also mentioned in removals.

Author: I have added a note in the removals part. Thanks

Bugfixes
--------

> - Multi-line newsfragments that ends with a code block will now have a newline inserted before appending the link to the issue, to avoid breaking formatting. (`#614 <https://github.com/twisted/towncrier/issues/614>`_)

Reviewer: ```suggestion
- Multi-line news fragments that end with a code block will now have a newline inserted before appending the link to the issue, to avoid breaking formatting. (`#614 <https://github.com/twisted/towncrier/issues/614>`_)
```

- Markdown header level is correctly inferred from ``title_format``. (`#695 <https://github.com/twisted/towncrier/issues/695>`_)

Improved Documentation
----------------------

- The documentation RST source files are now included in the sdist package. (`#672 <https://github.com/twisted/towncrier/issues/672>`_)
- Refactor the default markdown template to make it easier to understand, extend, and customize. (`#706 <https://github.com/twisted/towncrier/issues/706>`_)

Deprecations and Removals
-------------------------

> 

Author: ```suggestion

- Support for Python 3.8 was removed. (`#700 <https://github.com/twisted/towncrier/issues/700>`_)
```

- When no sections are present,
  the default Markdown template now renders the category headers as H2.

>   In previous versions it was rendered as H3. (`#663 <https://github.com/twisted/towncrier/issues/663>`_)

Reviewer: Not super related to the release but this section would make much more sense closer to the beginning of the change log.

>   In previous versions it was rendered as H3. (`#663 <https://github.com/twisted/towncrier/issues/663>`_)

Reviewer: Should this entry be under removals it's a feature / behavior change if you ask me.

Author: We can have a better name for `Deprecations and Removals` ... basically.
under `Deprecations and Removals` we have any "breaking changes"

We can look to update the template and use "Breaking changes"

Reviewer: @adiroiban yeah, I should probably propose updating the default templates. Over at @aio-libs and in other projects, I ended up having more accurate categories like contrib/packaging/downstream: https://yarl.aio-libs.org/en/latest/changes/#id100 / https://yarl.aio-libs.org/en/latest/contributing/guidelines/#alright-so-how-to-add-a-news-fragment / https://github.com/aio-libs/yarl/blob/b8d06ec9a7d95ac11437afd5e467c2c2bb1704d2/towncrier.toml#L18-L68.

Though, I wouldn't consider this all a part of the release PR. It's probably best to postpone until post-release.

Misc
----

> - `#667 <https://github.com/twisted/towncrier/issues/667>`_, `#669 <https://github.com/twisted/towncrier/issues/669>`_, `#679 <https://github.com/twisted/towncrier/issues/679>`_, `#680 <https://github.com/twisted/towncrier/issues/680>`_, `#682 <https://github.com/twisted/towncrier/issues/682>`_, `#701 <https://github.com/twisted/towncrier/issues/701>`_, `#702 <https://github.com/twisted/towncrier/issues/702>`_, `#713 <https://github.com/twisted/towncrier/issues/713>`_

Reviewer: Does this have to be a single bullet point?
```suggestion
`#667 <https://github.com/twisted/towncrier/issues/667>`_, `#669 <https://github.com/twisted/towncrier/issues/669>`_, `#679 <https://github.com/twisted/towncrier/issues/679>`_, `#680 <https://github.com/twisted/towncrier/issues/680>`_, `#682 <https://github.com/twisted/towncrier/issues/682>`_, `#701 <https://github.com/twisted/towncrier/issues/701>`_, `#702 <https://github.com/twisted/towncrier/issues/702>`_, `#713 <https://github.com/twisted/towncrier/issues/713>`_.
```

Author: This is how it's implemented now in the template and is how we did it in the past

-----

the ".misc" items might not have a description

Now... if you ask me, I would say that we should not list these items at all in the release notes

-----

But this is a discussion for a separate issue/PR

Reviewer: Agreed. This is separate. Though, I feel like misc items should be rendered with content, just like the others. If something isn't necessary in the change log, it should just not have a note to begin with.

Author: I like the .misc notes. ... but I agree they should not be rendered.

They are an explicit confirmation that the changes are not public.

It makes it easier to raise an errror in the case you accidentally forget to update the release notes for a public change.

------

We can look into this as part of updating the template that we user towncrier

We don't have to change the default template , towncrier can have a custom template ... but I think that it makese sense for towncrier itself to use the default template

I am happy to update the default template... but in a separate issue

.. towncrier release notes start

towncrier 25.8.0rc2 (2025-08-15)
================================

Features
--------

- Support was added for Mercurial SCM. (`#394 <https://github.com/twisted/towncrier/issues/394>`_)
- The :command:`towncrier check` command now has a ``--staged`` flag to inspect the files staged for commit when checking for a news fragment: useful in a pre-commit hook (`#676 <https://github.com/twisted/towncrier/issues/676>`_)
- When used with an :file:`pyproject.toml` file, when no explicit values are
  defined for ``[tool.towncrier.name|package]`` they will now fallback to
  the value of ``[project.name]``. (`#687 <https://github.com/twisted/towncrier/issues/687>`_)
- The `directory` configuration option is no longer required.

>   It can now be generated based on value from the `name` configuration.

Reviewer: It's probably a good idea to make the Sphinx config stricter about these any-role references:
```suggestion
- The ``directory`` configuration option is no longer required.
  It can now be generated based on value from the ``name`` configuration.
```
I've found that using `sphinxcontrib-towncrier` may also act as an early linter in PRs with more strictness in Sphinx. But we can talk about that separately.

Author: I am not sure what to say about 

```diff
- The `directory` configuration
+ The ``directory`` configuration
```

I prefer to be as relaxes as possible ... but we can update our documentation to help constributors understand why we prefer single backticks vs doubl backticks
